docs/adr/0004-code-structure.md

```diff
@@ -0,0 +1,70 @@
+# 4. Code Structure
+
+Date: 2018-06-14
+
+## Status
+
+Proposed
+
+## Context
+
+As the code rapidly grows we need a common structure and approach that will keep the code tidy and coherent.
+There are various aspects, the initial concern was the data access separation. The issue #140 initially talked about how this could be separated in terms of an ORM (Object-Relational Mapping) or a DAL (Data Access Layer).
+However, this discussion also soon covered the structure of the code in general and how to structure it for ease of development and best practise.
+
+### Rejected Options
+
+We considered the following options:
+
+* The Pubsweet Way - This was to define a base class "Model" and extend this for each entity. This was rejected as its another self made ORM but and has validations that we don't want.
+
+* Other ORMs : Waterline, sequelize, bookshelf - General purpose ORMs were rejected based on the experience that the time spent going up the learning curve and sifting through the docs when things go wrong is greater then simply rolling your own.
+
+* Query Builders : massive.js, Knex.js -These were not entirely ruled out, and were deemed compatible with a good code structure that would allow the database access to be encapsulated in a way that would allow a query builder to be used where necessary.
```

Comment:
Knex, yes, but Massive is not a query builder and it would not be easy to use it piecemeal.